axum/src/serve.rs

```diff
@@ -121,20 +121,23 @@ impl<M, S> Serve<M, S> {
     ///
     /// ```
     /// use axum::{Router, routing::get};
+    /// use std::{future::IntoFuture, sync::Arc};
+    /// use tokio::sync::Notify;
```

Comment:
Nit, can't we use just a oneshot channel for the example? As long as users don't need to see how to cancel from multiple places the whole `Arc` is unnecessary and it might be cleaner to have the semantics of being able to shutdown only once and having a dedicated side for producing and consuming the notification.

Reply:
The oneshot channel is fallible, so it has some noise too, but I still agree that it becomes a bit clearer that way. Updated.

Since `rx` yields `Result<(), _>` it would seem we can't just pass `rx` directly. Is there a way we could skip the `async move {...}` block and pass a future directly? It would seem we need to import `futures::future::FutureExt` to "eat" the `Result`? Would that still be preferable to using an async block? Are Axum users likely to have it in their `Cargo.toml` already?